zsh-proxy.plugin.zsh

```diff
@@ -147,17 +171,25 @@ __disable_proxy_git() {
 # NPM
 
 __enable_proxy_npm() {
-    npm config set proxy ${__ZSHPROXY_HTTP}
-    npm config set https-proxy ${__ZSHPROXY_HTTP}
-    yarn config set proxy ${__ZSHPROXY_HTTP} >/dev/null 2>&1
-    yarn config set https-proxy ${__ZSHPROXY_HTTP} >/dev/null 2>&1
+    if which npm >/dev/null; then
+      npm config set proxy ${__ZSHPROXY_HTTP}
+      npm config set https-proxy ${__ZSHPROXY_HTTP}
+      echo "- npm"
+    fi
+    if which yarn >/dev/null; then
+      yarn config set proxy ${__ZSHPROXY_HTTP} >/dev/null 2>&1
+      yarn config set https-proxy ${__ZSHPROXY_HTTP} >/dev/null 2>&1
+      echo "- yarn"
+    fi
 }
 
 __disable_proxy_npm() {
-    npm config delete proxy
-    npm config delete https-proxy
-    yarn config delete proxy >/dev/null 2>&1
-    yarn config delete https-proxy >/dev/null 2>&1
+    if which npm >/dev/null; then
+      npm config delete proxy
+      npm config delete https-proxy
+      yarn config delete proxy >/dev/null 2>&1
```

Comment:
Check for yarn is required here as well.

Reply:
Fixed